Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,371 @@
+            {
+                // convert follow target into spline local space, because SplineUtility works in spline local space
+                SplineUtility.GetNearestPoint(spline, 
+                    m_Spline.transform.InverseTransformPoint(FollowTargetPosition), out _, out normalizedSplinePosition, 
+                    m_AutoDolly.m_SearchResolution, m_AutoDolly.m_SearchIteration);
+                m_SplinePosition = SplineUtility.ConvertIndexUnit(spline, normalizedSplinePosition, 
+                    PathIndexUnit.Normalized, m_PositionUnits);
+                
+                // Apply the spline position offset
+                normalizedSplinePosition += m_AutoDolly.m_PositionOffset;
+            }
+
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                float maxUnit = 1; // we are always using normalized unit [0-1]
+                {
+                    float prev = m_PreviousNormalizedSplinePosition;
+                    float next = normalizedSplinePosition;
+                    if (spline.Closed && Mathf.Abs(next - prev) > maxUnit / 2)
+                    {
+                        if (next > prev)
+                            prev += maxUnit;
+                        else
+                            prev -= maxUnit;
+                    }
+                    m_PreviousNormalizedSplinePosition = prev;
+                    normalizedSplinePosition = next;
+                }
+
+                // Apply damping in the spline direction
+                float offset = m_PreviousNormalizedSplinePosition - normalizedSplinePosition;
+                offset = Damper.Damp(offset, m_Damping.z, deltaTime);
+                normalizedSplinePosition = m_PreviousNormalizedSplinePosition - offset;
+            }
+            m_PreviousNormalizedSplinePosition = normalizedSplinePosition;
+            m_Spline.Evaluate(normalizedSplinePosition, 
+                out var localPosition, out var localTangent, out var localUp);
+            Vector3 newCameraPos = localPosition;
+            if (Vector3.SqrMagnitude(localTangent) != 0)
+            {
+                m_CorrectedTangent = localTangent;
+            }
+            if (Vector3.SqrMagnitude(localUp) != 0)
+            {
+                m_CorrectedUp = localUp;
+            }
+            var newSplineOrientation = Quaternion.LookRotation(m_CorrectedTangent, m_CorrectedUp);
+
+            var rollExt = rollExtension;
+            if (rollExt != null && rollExt.enabled)
+            {
+                float roll = rollExt.RollOverride.Evaluate(spline, normalizedSplinePosition, 
+                    PathIndexUnit.Normalized, new UnityEngine.Splines.Interpolators.LerpFloat());
+                var rollRotation = Quaternion.AngleAxis(-roll, m_CorrectedTangent);
+                newSplineOrientation = Quaternion.LookRotation(m_CorrectedTangent, rollRotation * m_CorrectedUp);
+            }
```

Comment:
This is only used in some cameraUp modes, yet you are calculating it all the time.  Maybe it would make sense to move this logic into `GetCameraOrientationAtSplinePoint`, which knows whether or not it's needed.

Reply:
I am not sure if that's right. 

newSplineOrientation orientation affects the position offset of the camera (offsetX, Y, Z). I took TrackedDolly as a base, and there 
`var newSplineOrientation = Quaternion.LookRotation(m_CorrectedTangent, m_CorrectedUp);` affects the offset, then GetCameraOrientationAtSplinePoint only affects `curState.RawOrientation`.

It is possible that is not correct though.

Maybe GetCameraOrientationAtSplinePoint should affect the position offset?